filebeat/module/redis/log/ingest/pipeline.json

```diff
@@ -1,51 +1,81 @@
 {
-  "description": "Pipeline for parsing redis logs",
-  "processors": [
-    {
-      "grok": {
-        "field": "message",
-        "patterns": [
-          "(%{POSINT:process.pid:long}:%{CHAR:redis.log.role} )?%{REDISTIMESTAMP:redis.log.timestamp} %{REDISLEVEL:log.level} %{GREEDYDATA:message}",
-          "%{POSINT:process.pid:long}:signal-handler \\(%{POSINT:redis.log.timestamp}\\) %{GREEDYDATA:message}"
-        ],
-        "pattern_definitions": {
-          "CHAR": "[a-zA-Z]",
-          "REDISLEVEL": "[.\\-*#]"
+    "description": "Pipeline for parsing redis logs",
+    "processors": [
+        {
+            "grok": {
+                "field": "message",
+                "patterns": [
+                    "(%{POSINT:process.pid:long}:%{CHAR:redis.log.role} )?%{REDISTIMESTAMP:redis.log.timestamp} %{REDISLEVEL:log.level} %{GREEDYDATA:message}",
+                    "%{POSINT:process.pid:long}:signal-handler \\(%{POSINT:redis.log.timestamp}\\) %{GREEDYDATA:message}"
+                ],
+                "pattern_definitions": {
+                    "CHAR": "[a-zA-Z]",
+                    "REDISLEVEL": "[.\\-*#]"
+                }
+            }
+        },
+        {
+            "script": {
+                "source": "if (ctx.log.level == params.dot) {\n          ctx.log.level = params.debug;\n        } else if (ctx.log.level == params.dash) {\n          ctx.log.level = params.verbose;\n        } else if (ctx.log.level == params.asterisk) {\n          ctx.log.level = params.notice;\n        } else if (ctx.log.level == params.hash) {\n          ctx.log.level = params.warning;\n        }",
+                "params": {
+                    "dot": ".",
+                    "debug": "debug",
+                    "dash": "-",
+                    "verbose": "verbose",
+                    "asterisk": "*",
+                    "notice": "notice",
+                    "hash": "#",
+                    "warning": "warning"
+                }
+            }
+        },
+        {
+            "script": {
+                "lang": "painless",
+                "source": "if (ctx.redis.log.role == params.master_abbrev) {\n          ctx.redis.log.role = params.master;\n        } else if (ctx.redis.log.role == params.slave_abbrev) {\n          ctx.redis.log.role = params.slave;\n        } else if (ctx.redis.log.role == params.child_abbrev) {\n          ctx.redis.log.role = params.child;\n        } else if (ctx.redis.log.role == params.sentinel_abbrev) {\n          ctx.redis.log.role = params.sentinel;\n        }\n        ",
+                "params": {
+                    "master_abbrev": "M",
```

Comment:
It seems odd that we have to extract all these constants which are only used once but I don't think there is a better way. Perhaps @jakelandis can comment?